### PR TITLE
Resolve Pr comments

### PR DESCRIPTION
**Pull request by anmolbansal144**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -99,11 +99,13 @@
             <groupId>org.apache.kafka</groupId>
             <artifactId>connect-api</artifactId>
             <version>${kafka.version}</version>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>org.apache.kafka</groupId>
             <artifactId>connect-json</artifactId>
             <version>${kafka.version}</version>
+            <scope>provided</scope>
```
Comment (fahadhasher) [inline pom.xml:108]: why this and below dependencies are scoped provided?

File: src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/KustoSinkTask.java [hunk 2]
```diff
@@ -221,6 +225,20 @@ void validateTableMappings(KustoSinkConfig config) {
         }
     }
 
+    private static boolean isIngestorRole(JSONObject testMapping, Client engineClient) throws JSONException {
```
Comment (fahadhasher) [inline src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/KustoSinkTask.java:228]: why static?

File: src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/KustoSinkTask.java [hunk 3]
```diff
@@ -193,9 +193,13 @@ void validateTableMappings(KustoSinkConfig config) {
             Client engineClient = createKustoEngineClient(config);
             if (config.getTopicToTableMapping() != null) {
                 JSONArray mappings = new JSONArray(config.getTopicToTableMapping());
-                for (int i = 0; i < mappings.length(); i++) {
-                    JSONObject mapping = mappings.getJSONObject(i);
-                    validateTableAccess(engineClient, mapping, config, databaseTableErrorList, accessErrorList);
+                if(mappings.length()>0) {
+                    if(isIngestorRole(mappings.getJSONObject(0), engineClient)) {
```
Comment (fahadhasher) [inline src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/KustoSinkTask.java:197]: if it's not ingestor role then we should perform validations, correct?

File: src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/KustoSinkTask.java [hunk 4]
```diff
@@ -193,9 +193,13 @@ void validateTableMappings(KustoSinkConfig config) {
             Client engineClient = createKustoEngineClient(config);
             if (config.getTopicToTableMapping() != null) {
                 JSONArray mappings = new JSONArray(config.getTopicToTableMapping());
-                for (int i = 0; i < mappings.length(); i++) {
-                    JSONObject mapping = mappings.getJSONObject(i);
-                    validateTableAccess(engineClient, mapping, config, databaseTableErrorList, accessErrorList);
+                if(mappings.length()>0) {
```
Comment (fahadhasher) [inline src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/KustoSinkTask.java:196]:
```suggestion
                if(mappings.length() > 0) {
```

File: src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/KustoSinkTask.java (hunk 3 above)
Comment (SanchayGupta1197) [reply src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/KustoSinkTask.java:197]: yes

Comment (SanchayGupta1197) [reply src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/KustoSinkTask.java:197]: had checked for the check that will fail on the ingestor role but will pass on the viewer/admin roles.

File: pom.xml (hunk 1 above)
Comment (anmolbansal144) [reply pom.xml:108]: ohad said to us : https://github.com/Azure/kafka-sink-azure-kusto/pull/31#discussion_r452271344

Comment (anmolbansal144) [reply pom.xml:108]: ohad said to us to change the scope of all connect-* dependency to provided

File: src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/FileWriter.java [hunk 5]
```diff
@@ -178,8 +175,8 @@ private void dumpFile() throws IOException {
     }
 
     public void rollback() throws IOException {
-        if (outputStream != null) {
-            outputStream.close();
+        if (countingStream != null) {
+            countingStream.close();
             if (currentFile != null && currentFile.file != null) {
                 dumpFile();
             }
```
Comment (Sumanthonnavar) [inline src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/FileWriter.java:182]: Indentation

File: src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/FileWriter.java [hunk 6]
```diff
@@ -291,31 +288,5 @@ else if ((record.valueSchema() != null) && (record.valueSchema().type() == Schem
             throw new ConnectException(String.format("Invalid Kafka record format, connector does not support %s format. This connector supports Avro, Json with schema, Json without schema, Byte, String format. ",record.valueSchema().type()));
         }
     }
-
-    private class CountingOutputStream extends FilterOutputStream {
```
Comment (Sumanthonnavar) [inline src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/FileWriter.java:295]: Why did we remove this?

File: src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/CountingOutputStream.java [hunk 7]
```diff
@@ -0,0 +1,37 @@
+package com.microsoft.azure.kusto.kafka.connect.sink;
+
+import java.io.FilterOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+
+public class CountingOutputStream extends FilterOutputStream {
+  public long numBytes = 0;
+  public OutputStream outputStream;
```
Comment (Sumanthonnavar) [inline src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/CountingOutputStream.java:9]: Make it private

File: README.md [hunk 8]
```diff
@@ -74,8 +74,8 @@ flush.interval.ms=300000
 
 behavior.on.error=FAIL
 
-dlq.bootstrap.servers=localhost:9092
-dlq.topic.name=test-topic-error
+misc.deadletterqueue.bootstrap.servers=localhost:9092
```
Comment (fahadhasher) [inline README.md:77]: @SanchayGupta1197 maybe add the connect dlq props as well.

File: README.md [hunk 9]
```diff
@@ -112,17 +112,13 @@ KafkaTest | count
...
+`csv`, `json`, `avro`, `apacheAvro`, `tsv`, `scsv`, `sohsv`, `psv`, `txt`.
 
-> Note - `avro`, `apacheAvro`, `parquet` and `orc` files are sent each record (file) separately without aggregation, and are expected to be sent as a byte array containing the full file.
+> Note - `avro` and `apacheAvro`files are sent each record (file) separately without aggregation, and are expected to be sent as a byte array containing the full file.
 > 
 >Use `value.converter=org.apache.kafka.connect.converters.ByteArrayConverter`
 
 
-#### Supported compressions
-Kusto Kafka connector can get compressed data, this can be specified in the topics_mapping in the configuration under
```
Comment (fahadhasher) [inline README.md:123]: Maybe write : All the records processed by the Connector(except for records having schema as bytearray) are compressed after flushing them into a file before ingesting it into Kusto.

File: connect-kusto-sink.properties [hunk 10]
```diff
@@ -17,8 +17,8 @@ tasks.max=1
 
 #behavior.on.error=FAIL
 
-#dlq.bootstrap.servers=localhost:9092
-#dlq.topic.name=test-topic-error
+#misc.deadletterqueue.bootstrap.servers=localhost:9092
+#misc.deadletterqueue.topic.name=test-topic-error
```
Comment (fahadhasher) [inline connect-kusto-sink.properties:21]: add connect dlq props.

File: src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/KustoSinkConfig.java [hunk 11]
```diff
@@ -90,16 +90,16 @@
         + "while processing records or ingesting records in Kusto table, available in connect logs.";
     private static final String KUSTO_BEHAVIOR_ON_ERROR_DISPLAY = "Behavior On Error";
     
-    static final String KUSTO_DLQ_BOOTSTRAP_SERVERS_CONF = "dlq.bootstrap.servers";
+    static final String KUSTO_DLQ_BOOTSTRAP_SERVERS_CONF = "misc.deadletterqueue.bootstrap.servers";
     private static final String KUSTO_DLQ_BOOTSTRAP_SERVERS_DOC = "Configure this list to Kafka broker's address(es) "
-        + "to which the Connector should write failed records to. "
+        + "to which the Connector should write records failed due to network interruptions or unavailability of Kusto cluster to. "
```
Comment (fahadhasher) [inline src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/KustoSinkConfig.java:95]:
```suggestion
        + "to which the Connector should write records failed due to network interruptions or unavailability of Kusto cluster. "
```

File: src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/KustoSinkConfig.java [hunk 12]
```diff
@@ -90,16 +90,16 @@
...
         + "This list should be in the form host-1:port-1,host-2:port-2,…host-n:port-n. ";
-    private static final String KUSTO_DLQ_BOOTSTRAP_SERVERS_DISPLAY = "Dead-Letter Queue Bootstrap Servers";
+    private static final String KUSTO_DLQ_BOOTSTRAP_SERVERS_DISPLAY = "Miscellaneous Dead-Letter Queue Bootstrap Servers";
     
-    static final String KUSTO_DLQ_TOPIC_NAME_CONF = "dlq.topic.name";
+    static final String KUSTO_DLQ_TOPIC_NAME_CONF = "misc.deadletterqueue.topic.name";
     private static final String KUSTO_DLQ_TOPIC_NAME_DOC = "Set this to the Kafka topic's name "
-        + "to which the failed records are to be sinked.";
-    private static final String KUSTO_DLQ_TOPIC_NAME_DISPLAY = "Dead-Letter Queue Topic Name";
+        + "to which the records failed due to network interruptions or unavailability of Kusto cluster are to be sinked.";
```
Comment (fahadhasher) [inline src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/KustoSinkConfig.java:101]: keep this line consistent.

File: src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/TopicPartitionWriter.java [hunk 13]
```diff
@@ -157,14 +154,7 @@ String getFilePath(@Nullable Long offset) {
...
 
-        String compressionExtension = "";
-        if (shouldCompressData(ingestionProps, null) || eventDataCompression != null) {
-            if(eventDataCompression != null) {
-                compressionExtension = "." + eventDataCompression.toString();
-            } else {
-                compressionExtension = ".gz";
-            }
-        }
+        String compressionExtension = ".gz";
```
Comment (fahadhasher) [inline src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/TopicPartitionWriter.java:157]: constant?

File: src/test/java/com/microsoft/azure/kusto/kafka/connect/sink/FileWriterTest.java [hunk 14]
```diff
@@ -258,37 +258,41 @@ public void offsetCheckByInterval() throws InterruptedException, IOException {
...
-        fileWriter.close();
-        Assert.assertEquals(Objects.requireNonNull(folder.listFiles()).length, 1);
-    }
+/*
+ Test commented as it required shouldCompressData as false which is actually true while running
+ the connector as we are compression all files as gzip.
+*/
+
+//    @Test
+//    public void testFileWriterCompressed() throws IOException {
```
Comment (fahadhasher) [inline src/test/java/com/microsoft/azure/kusto/kafka/connect/sink/FileWriterTest.java:267]: why have we removed a test and commented out the other one?

File: README.md (hunk 8 above)
Comment (SanchayGupta1197) [reply README.md:77]: done

Comment (SanchayGupta1197) [reply README.md:77]: done

File: connect-kusto-sink.properties (hunk 10 above)
Comment (SanchayGupta1197) [reply connect-kusto-sink.properties:21]: done

File: README.md (hunk 9 above)
Comment (SanchayGupta1197) [reply README.md:123]: done

File: src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/KustoSinkConfig.java (hunk 12 above)
Comment (SanchayGupta1197) [reply src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/KustoSinkConfig.java:101]: done

File: src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/TopicPartitionWriter.java (hunk 13 above)
Comment (SanchayGupta1197) [reply src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/TopicPartitionWriter.java:157]: done

**Pull request merged**